### PR TITLE
[User][Customer] Clean up of components and bundles

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Related tickets | -
| License         | MIT

- [x] User component
- [x] User bundle
- [x] Customer component
- [x] Customer bundle

### CODE REVIEW

File: src/Sylius/Component/User/Model/User.php [hunk 1]
```diff
@@ -499,8 +497,6 @@ public function addOAuthAccount(UserOAuthInterface $oauth)
     }
 
     /**
-     * Returns username.
-     *
      * @return string
      */
     public function __toString()
```
Comment (pamil) [inline src/Sylius/Component/User/Model/User.php]: Could be moved to the top of this class.

File: src/Sylius/Component/User/Security/UserPbkdf2PasswordEncoder.php [hunk 2]
```diff
@@ -68,7 +68,8 @@ public function __construct($algorithm = 'sha512', $encodeHashAsBase64 = true, $
     /**
      * {@inheritdoc}
      *
-     * @throws \LogicException when the algorithm is not supported
+     * @throws \InvalidArgumentException
+     * @throws \LogicException
```
Comment (pamil) [inline src/Sylius/Component/User/Security/UserPbkdf2PasswordEncoder.php]: Reason should be reverted.

File: src/Sylius/Component/User/Security/UserPbkdf2PasswordEncoder.php [hunk 3]
```diff
@@ -81,12 +82,16 @@ public function encode(CredentialsHolderInterface $user)
...
      */
     protected function encodePassword($plainPassword, $salt)
     {
         if ($this->isPasswordTooLong($plainPassword)) {
-            throw new \InvalidArgumentException('Too long password.');
+            throw new \InvalidArgumentException(sprintf(
+                'The password must be at most %d characters long',
+                self::MAX_PASSWORD_LENGTH)
```
Comment (pamil) [inline src/Sylius/Component/User/Security/UserPbkdf2PasswordEncoder.php]: Could we move this `)` to the next line, next to another `)`? :>

File: src/Sylius/Bundle/CustomerBundle/Form/Type/CustomerProfileType.php [hunk 4]
```diff
@@ -25,28 +28,28 @@ class CustomerProfileType extends AbstractResourceType
...
-            ->add('firstName', 'text', [
+            ->add('firstName', TextType::class, [
                 'label' => 'sylius.form.customer.first_name',
             ])
-            ->add('lastName', 'text', [
+            ->add('lastName', TextType::class, [
                 'label' => 'sylius.form.customer.last_name',
             ])
-            ->add('email', 'email', [
+            ->add('email', TextType::class, [
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CustomerBundle/Form/Type/CustomerProfileType.php]: Shouldn't be `email`?

File: src/Sylius/Bundle/CustomerBundle/Resources/translations/flashes.en.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+sylius:
+    customer:
+        register: Thank you for registering, check your email to verify your account.
```
Comment (tuka217) [inline src/Sylius/Bundle/CustomerBundle/Resources/translations/flashes.en.yml]: Shouldn't we use apostrophes?

File: src/Sylius/Bundle/UserBundle/spec/Provider/EmailProviderSpec.php [hunk 6]
```diff
@@ -50,16 +52,19 @@ function it_supports_sylius_user_model()
         $this->supportsClass(User::class)->shouldReturn(true);
     }
 
-    function it_loads_user_by_email($userRepository, $canonicalizer, User $user)
-    {
+    function it_loads_user_by_email(
+        UserRepositoryInterface $userRepository,
+        CanonicalizerInterface $canonicalizer,
+        User $user
```
Comment (tuka217) [inline src/Sylius/Bundle/UserBundle/spec/Provider/EmailProviderSpec.php]: `UserInterface`

File: src/Sylius/Bundle/UserBundle/spec/Provider/EmailProviderSpec.php [hunk 7]
```diff
@@ -50,16 +52,19 @@ function it_supports_sylius_user_model()
...
+    ) {
         $canonicalizer->canonicalize('test@user.com')->willReturn('test@user.com');
 
         $userRepository->findOneByEmail('test@user.com')->willReturn($user);
 
         $this->loadUserByUsername('test@user.com')->shouldReturn($user);
     }
 
-    function it_updates_user_by_user_name($userRepository, User $user)
+    function it_updates_user_by_user_name(UserRepositoryInterface $userRepository, User $user)
```
Comment (tuka217) [inline src/Sylius/Bundle/UserBundle/spec/Provider/EmailProviderSpec.php]: Same here

File: src/Sylius/Bundle/UserBundle/spec/Provider/UsernameProviderSpec.php [hunk 8]
```diff
@@ -50,16 +51,19 @@ function it_supports_sylius_user_model()
         $this->supportsClass(User::class)->shouldReturn(true);
     }
 
-    function it_loads_user_by_user_name($userRepository, $canonicalizer, User $user)
-    {
+    function it_loads_user_by_user_name(
+        UserRepositoryInterface $userRepository,
+        CanonicalizerInterface $canonicalizer,
+        User $user
```
Comment (tuka217) [inline src/Sylius/Bundle/UserBundle/spec/Provider/UsernameProviderSpec.php]: UserInterface

File: src/Sylius/Component/User/Model/UserInterface.php [hunk 9]
```diff
@@ -97,18 +97,19 @@ public function getPasswordResetToken();
     public function setPasswordResetToken($passwordResetToken);
 
     /**
-     * Sets the timestamp that the user requested a password reset.
-     *
+     * @return null|\DateTime
```
Comment (tuka217) [inline src/Sylius/Component/User/Model/UserInterface.php]: Maybe `\DateTime|null`?

File: src/Sylius/Bundle/UserBundle/EventListener/UserDeleteListener.php [hunk 10]
```diff
@@ -48,19 +46,22 @@ public function __construct(TokenStorageInterface $tokenStorage, SessionInterfac
...
     public function deleteUser(GenericEvent $event)
     {
         $user = $event->getSubject();
-
         if (!$user instanceof UserInterface) {
             throw new UnexpectedTypeException(
                 $user,
                 UserInterface::class
             );
         }
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/UserBundle/EventListener/UserDeleteListener.php]: `Assert::instanceOf`

File: src/Sylius/Component/User/Security/UserPbkdf2PasswordEncoder.php [hunk 11]
```diff
@@ -52,10 +52,10 @@ class UserPbkdf2PasswordEncoder implements UserPasswordEncoderInterface
...
 
     /**
-     * @param string $algorithm          The digest algorithm to use
-     * @param bool   $encodeHashAsBase64 Whether to base64 encode the password hash
-     * @param int    $iterations         The number of iterations to use to stretch the password hash
-     * @param int    $length             Length of derived key to create
+     * @param string $algorithm
+     * @param bool $encodeHashAsBase64
+     * @param int $iterations
+     * @param int $length
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/User/Security/UserPbkdf2PasswordEncoder.php]: The above can be useful.

**Comment by michalmarcinkowski** [comment]
Thank you Janek!

**Pull request merged**